The build failed with the following errors:

```
> Task :x-pack:plugin:watcher:compileTestJava
/.../elasticsearch-pull-request/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleServiceTests.java:314: error: cannot find symbol
            .settings(settings(IndexVersion.current()).put(IndexMetadata.INDEX_FORMAT_SETTING.getKey(), 6)) // the internal index format,
                               ^
  symbol:   variable IndexVersion
  location: class WatcherLifeCycleServiceTests
/.../elasticsearch-pull-request/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleServiceTests.java:325: error: cannot find symbol
            .nodes(nodes.withMasterNodeId("node_1"))
                        ^
  symbol:   method withMasterNodeId(String)
  location: variable nodes of type DiscoveryNodes
/.../elasticsearch-pull-request/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleServiceTests.java:330: error: cannot find symbol
            .nodes(nodes.withMasterNodeId("node_2"))
                        ^
  symbol:   method withMasterNodeId(String)
  location: variable nodes of type DiscoveryNodes
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
3 errors
```

The first error indicates that the test depends on https://github.com/elastic/elasticsearch/pull/94827, which is not back-ported to the 7.17 branch. But `IndexVersion.current()` could be rewritten with `Version.CURRENT`.

The second and third errors indicate that the test depends on https://github.com/elastic/elasticsearch/pull/96836, which is also not back-ported to the 7.17. branch. But `nodes.withMasterNodeId("X")` could be rewritten with `DiscoveryNodes.builder(nodes).masterNodeId("X")`.

--- a/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleServiceTests.java
+++ b/x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleServiceTests.java
@@ -311,7 +311,7 @@ public class WatcherLifeCycleServiceTests extends ESTestCase {
         RoutingTable routingTable = RoutingTable.builder().add(indexRoutingTableBuilder.build()).build();
 
         IndexMetadata.Builder indexMetadataBuilder = IndexMetadata.builder(Watch.INDEX)
-            .settings(settings(IndexVersion.current()).put(IndexMetadata.INDEX_FORMAT_SETTING.getKey(), 6)) // the internal index format,
+            .settings(settings(Version.CURRENT).put(IndexMetadata.INDEX_FORMAT_SETTING.getKey(), 6)) // the internal index format,
             // required
             .numberOfShards(1)
             .numberOfReplicas(0);
@@ -322,12 +322,12 @@ public class WatcherLifeCycleServiceTests extends ESTestCase {
 
         ClusterState emptyState = ClusterState.builder(new ClusterName("my-cluster")).nodes(nodes).metadata(metadata).build();
         ClusterState stateWithMasterNode1 = ClusterState.builder(new ClusterName("my-cluster"))
-            .nodes(nodes.withMasterNodeId("node_1"))
+            .nodes(DiscoveryNodes.builder(nodes).masterNodeId("node_1"))
             .metadata(metadata)
             .routingTable(routingTable)
             .build();
         ClusterState stateWithMasterNode2 = ClusterState.builder(new ClusterName("my-cluster"))
-            .nodes(nodes.withMasterNodeId("node_2"))
+            .nodes(DiscoveryNodes.builder(nodes).masterNodeId("node_2"))
             .metadata(metadata)
             .routingTable(routingTable)
             .build();

